File: solutions/typescript/resistor-color/1/resistor-color.ts
enum ResistorColor {
  Black = "black",
  Brown = "brown",
  Red = "red",
  Orange = "orange",
  Yellow = "yellow",
  Green = "green",
  Blue = "blue",
  Violet = "violet",
  Grey = "grey",
  White = "white",
}

const resistorValues: Record<ResistorColor, number> = {
  [ResistorColor.Black]: 0,
  [ResistorColor.Brown]: 1,
  [ResistorColor.Red]: 2,
  [ResistorColor.Orange]: 3,
  [ResistorColor.Yellow]: 4,
  [ResistorColor.Green]: 5,
  [ResistorColor.Blue]: 6,
  [ResistorColor.Violet]: 7,
  [ResistorColor.Grey]: 8,
  [ResistorColor.White]: 9,
};

export const colorCode = (color: ResistorColor) => resistorValues[color];

export const COLORS = Object.values(ResistorColor);

File: solutions/python/resistor-color/1/resistor_color.py
from enum import Enum

class ColorCode(Enum):
    BLACK  = 0
    BROWN  = 1
    RED    = 2
    ORANGE = 3
    YELLOW = 4
    GREEN  = 5
    BLUE   = 6
    VIOLET = 7
    GREY   = 8
    WHITE  = 9

def color_code(color: str) -> int:
    try:
        return ColorCode[color.upper()].value
    except KeyError:
        raise ValueError(f"Invalid color: {color!r}")

def colors() -> list[str]:
    return [c.name.lower() for c in ColorCode]
